pkg/integrations/integrations.go

```diff
@@ -0,0 +1,53 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/k8sgpt-ai/k8sgpt-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	backstageLabelKey = "backstage.io/kubernetes-id"
+)
+
+func BackstageLabel(ctx context.Context, c client.Client, result v1alpha1.ResultSpec) (map[string]string, error) {
+
+	namespace, resourceName, _ := strings.Cut(result.Name, "/")
+	m, err := cmdutil.NewFactory(genericclioptions.NewConfigFlags(true)).ToRESTMapper()
```

Comment:
This will create as new client each call, which will be many times in large clusters, the issue with this ( I believe ) is that it's not a singleton and will increase resource usage over time.

I would suggestion we make the `integrations` or `support` as you suggested, file use a struct

```
type Integrations struct {
  restClient *rest.Client
}

func (i *Integrations) LabelBackstageResource(result v1alpha1.ResultSpec) (map[string]string, error) {} 

func NewIntegrations(ctx context.Context, c client.Client) {} 
```

This would mean you can initialise the rest client but also to store the passed in `client.Client`